### PR TITLE
Add GHA workflow

### PR DESCRIPTION
**Pull request by Sushisource**
Need to merge this to main before it can be tested. Thanks GHA.

### CODE REVIEW

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: Nightly Tests
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    - cron:  '00 19 * * *'
+
+jobs:
+  stress-test:
+    runs-on: self-hosted
```
Comment (bergundy) [inline .github/workflows/nightly.yml:9]: This isn't a good name for the runner.
How do I distinguish this from lets say a windows or mac runner?

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        with:
+          node-version: ${{ matrix.node }}
+      # Don't build during install phase since we're going to explicitly build
+      - run: npm ci --ignore-scripts
+      - run: npm run build
+        env:
+          BUILD_CORE_RELEASE: true
+
+      # docker-compose
+      - name: Checkout docker-compose repo
```
Comment (bergundy) [inline .github/workflows/nightly.yml:27]: Maybe just use temporalite, the default compose setup is heavy and slow?

**Review by bergundy — APPROVED** [review]
Had some comments but those can be address in the next PR when you implement the actual tests

File: .github/workflows/nightly.yml (hunk 1 above)
Comment (Sushisource) [reply .github/workflows/nightly.yml:9]: I can't control that name. You have to use that label, and you can use additional ones, but they're all linux.

Comment (bergundy) [reply .github/workflows/nightly.yml:9]: Right, you'd need to use an array of labels.
For the M1 Mac, we have: [self-hosted, MacOS, arm64]

**Pull request merged**